mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -830,17 +830,617 @@ LogicalResult ThreadwiseWriteAllRewritePattern::matchAndRewrite(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// BlockwiseReduce lowering.
+//===----------------------------------------------------------------------===//
+
+struct BlockwiseReduceRewritePattern
+    : public OpConversionPattern<BlockwiseReduceOp> {
+  using OpConversionPattern<BlockwiseReduceOp>::OpConversionPattern;
+
+  int64_t calculateNonReductionDimProduct(ArrayRef<int64_t> toReduceShape,
+                                          int64_t axis) const {
+    int64_t dimProduct = 1;
+    for (size_t i = 0; i < toReduceShape.size(); i++) {
+      if (i != (size_t)axis) {
+        dimProduct *= toReduceShape[i];
+      }
+    }
+    return dimProduct;
+  }
+
+  // This function will append views to target a flat LDS buffer
+  // where non-reduction dims are laid contigously as they are expected
+  // function on parallel.
+  ArrayAttr createLDSWorkspaceView(Location loc, PatternRewriter &rewriter,
+                                   ArrayAttr regTensorView, int64_t reduceAxis,
+                                   bool makeRDimZero = false) const {
+
+    TransformMapAttr lowestTr =
```

Comment:
Could you instead `getShape()` the underlying object instead of taking the output shape of the last transform? This feels unnecessary.

Reply:
I dont have such an object/op -- just a bunch of manipulated transforms (basically bid stripped)